openedx/features/course_experience/api/v1/views.py

```diff
@@ -37,3 +42,48 @@ def reset_course_deadlines(request):
         return Response({'message': 'Deadlines successfully reset.'})
     except Exception:
         raise UnableToResetDeadlines
+
+
+class CourseDeadlinesMobileView(RetrieveAPIView):
+    """
+    **Use Cases**
+
+        Request course deadline info for mobile
+
+    **Example Requests**
+
+        GET api/course_experience/v1/course_deadlines_info/{course_key}
+
+    **Response Values**
+
+        Body consists of the following fields:
+
+        dates_banner_info: (obj)
+            missed_deadlines: (bool) Whether the user has missed any graded content deadlines for the given course.
+            missed_gated_content: (bool) Whether the user has missed any gated content for the given course.
+            content_type_gating_enabled: (bool) Whether content type gating is enabled for this enrollment.
+            verified_upgrade_link: (str) The URL to ecommerce IDA for purchasing the verified upgrade.
+
+    **Returns**
+
+        * 200 on success with above fields.
+        * 401 if the user is not authenticated.
+        * 404 if the course is not available or cannot be seen.
+    """
+
+    authentication_classes = (
+        JwtAuthentication,
+        BearerAuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
+    permission_classes = (IsAuthenticated,)
+    serializer_class = CourseDeadlinesMobileSerializer
+
+    def get(self, request, *args, **kwargs):
+        course_key_string = kwargs.get('course_key_string')
+        course_key = CourseKey.from_string(course_key_string)
+        get_course_with_access(request.user, 'load', course_key)
```

Comment:
I wonder if this is the best way to check for course existence. Might be worth a quick question to... someone, but honestly, I'm fine with it.

Reply:
So, I missed this, but I think you probably would be better served by `get_course_overview_with_access`, since you don't actually need any of the course content.